rotkehlchen/db/dbhandler.py

```diff
@@ -2660,21 +2661,34 @@ def get_rotkehlchen_premium(self) -> Optional[PremiumCredentials]:
         # else
         return None
 
-    def get_netvalue_data(self, from_ts: Timestamp) -> Tuple[List[str], List[str]]:
+    def get_netvalue_data(
+        self,
+        from_ts: Timestamp,
+        include_nfts: bool = True,
+    ) -> Tuple[List[str], List[str]]:
         """Get all entries of net value data from the DB"""
         cursor = self.conn.cursor()
         # Get the total location ("H") entries in ascending time
         query = cursor.execute(
             f'SELECT time, usd_value FROM timed_location_data '
             f'WHERE location="H" AND time >= {from_ts} ORDER BY time ASC;',
         )
-
+        historical_data = query.fetchall()
+        if not include_nfts:
+            nft_balances = cursor.execute(
+                f'SELECT time, SUM(usd_value) FROM timed_balances WHERE time >= ? '
+                f'AND currency LIKE "{NFT_DIRECTIVE}%" GROUP BY time',
+                (from_ts,),
+            )
+            nft_values = {time: value for time, value in nft_balances}
         data = []
         times_int = []
-        for entry in query:
+        for entry in historical_data:
```

Comment:
Here you are also looping the `historical_data` 2 times.

1. in the fetchall()
2. in the loop

Just remove  `historical_data = query.fetchall()` and replace `for entry in historical_data` with `for entry in query`